src/styles/index.scss

```diff
@@ -1,3 +1,67 @@
+$start-day: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+
+html {
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
 body {
   margin: 0;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  padding: 0 10px;
+  width: calc(7 * 100px + 6 * 1px);
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  .calendar__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 100px;
+    height: 100px;
+    box-sizing: border-box;
+    border: 1px solid black;
+    background-color: #eee;
+    transition-duration: 0.5s;
+    cursor: pointer;
+
+    &:hover {
+      background-color: #ffbfcb;
+      transform: translate(0, -20px);
+    }
+  }
+
+  @for $numberday from 1 through 31 {
+    .calendar__day:nth-child(n + #{$numberday})::before {
+      content: '#{$numberday}';
+    }
+  }
+}
+
+@each $day, $i in $start-day {
+  .calendar--start-day-#{$day} .calendar__day:first-child {
+    margin-left: calc(#{$i} * 100px + #{$i * 1px});
```

Comment:
The calculation for `margin-left` should ensure it accounts for the gap between days. Currently, it multiplies the index by 100px and adds the index multiplied by 1px, which may not correctly reflect the intended layout if the gap is different from 1px. Verify that this calculation aligns with your design intentions.